arch/arm/src/sama5/sam_adc.c

```diff
@@ -865,7 +935,7 @@ static void sam_adc_endconversion(void *arg)
   int ret;
 
   DEBUGASSERT(priv != NULL);
-  ainfo("pending=%08x\n", priv->pending);
+  ainfo("pending=%08lx\n", priv->pending);
```

Comment:
PRIx32

Reply:
I come across so many of these in existing code....

Reply:
But this one is already fixed...its not in master as of now?

Reply:
We need to fix this

Reply:
still has %08lx in master, don't know why it shows wrong for you

Reply:
I mean, that we need to change to PRIx32

Reply:
I've never used that macro. Is it preferred for NuttX? There are plenty of instances of %08x and %08lx littered through the codebase.

Reply:
That is a standard way of printing stdint.h types. The print macro are located in inttypes.h

Reply:
Been looking it up and I think inttypes.h is C99? Aren't we supposed to use C89?

Reply:
Or, at least, there's no "need" to use C99 features in arch files even if we can?

Reply:
NuttX has some C99 extensions generated. Like stdbool.h or stdint.h
Since those are not compiler dependant and can be used for all archs

Reply:
Yes, but I want to differentiate between errors that I should fix, or preferences. I think this is in the preference category, so I don't want to spend time fixing issues in other peoples code if they're not actually an issue or error.

Reply:
Here is some reference discussion, so you can better understand what I'm taking about https://github.com/apache/nuttx/pull/7476#issuecomment-1296318701

Just never mind, I will fix it some time later when will get some free time.

Reply:
That discussion is slightly beyond me, but I'm trying to understand if using %08lx is actually WRONG rather than just a difference in approach. We *can* use C99 in arch but don't HAVE to? I have always used the %08x type formatters, probably because I learnt C back in the 1980's (K&R!).

I am not trying to be difficult, but need to make sure that any PR I submit 100% meets the rules, ideally first time! If it's "just" a suggestion that it is "better" to do something using a different method I am always happy to listen and learn, but need to know if that's what it is.

Reply:
> That discussion is slightly beyond me, but I'm trying to understand if using %08lx is actually WRONG rather than just a difference in approach.

no prefix mean int, l prefix is for long int. But int32_t may map to int or long int which depends on arch and toolchain, so it's wrong to either use l prefix or no prefix for int32_t. To make programmer life easier, the standard introduce PRI[d|u|x]32 as the int32_t format specifier.

> We _can_ use C99 in arch but don't HAVE to? I have always used the %08x type formatters, probably because I learnt C back in the 1980's (K&R!).
> 

So the rule is simple:
int: no prefix
long: l prefix
long long: ll prefix
int32_t: PRI[o||d|u|x]32
Here has some discussion:
https://stackoverflow.com/questions/3168275/printf-format-specifiers-for-uint32-t-and-size-t?rq=1

Reply:
Thanks! Because the SAMA5s are 32 bit devices (and I have only ever written serious code for 32 bit devices) the %08lx is technically correct I think? But I understand that anything that makes code clearer to any coder is a good thing. I will adopt this and VERY much appreciate the clear explanation :)